File: comparisons/mapreduce/mapreduce.fs

let somaDosParesAoQuadrado = 
    [1..100]
    |> Seq.filter (fun x     -> x % 2 = 0 )
    |> Seq.map    (fun x     -> x * x)
    |> Seq.reduce (fun acc x -> acc + x)

printfn "%A" somaDosParesAoQuadrado

File: comparisons/mapreduce/mapreduce.py

somaDosParesAoQuadrado =            \
    reduce(lambda acc, x: acc + x,  \
    map(lambda x: x ** 2,           \
    filter(lambda x: x % 2 == 0,    \
    range(1, 101))))

print somaDosParesAoQuadrado